### PR TITLE
SO-2818: clean up component types

### PR DESCRIPTION
**Pull request by cmark**
This PR changes the core metadata model to have a single String ID that represents a terminology component as opposed to having three different types and methods to access/use them.

### CODE REVIEW

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/branch/compare/CompareResultsDsvExporter.java [hunk 1]
```diff
@@ -254,7 +254,7 @@
 //			this.changeKind = changeKind;
 //			this.component = component;
 //			this.label = label;
-//			this.componentType = TerminologyRegistry.INSTANCE.getTerminologyComponentByShortId(component.getTerminologyComponentId()).name();
+//			this.componentType = TerminologyRegistry.INSTANCE.getTerminologyComponentByShortId(component.getComponentType()).name();
```
Comment (apeteri) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/branch/compare/CompareResultsDsvExporter.java:257]: The call to `TerminologyRegistry` should no longer be needed, I think 👼

File: fhir/com.b2international.snowowl.fhir.rest.tests/src/com/b2international/snowowl/fhir/rest/tests/conceptmap/SnomedConceptMapRestTest.java [hunk 2]
```diff
@@ -223,8 +222,7 @@ public void getExtendedMapTypeConceptMapTest() {
 	public void conceptMapsSummaryTest() throws Exception {
 		
 		String simpleMapTypeRefsetId = mapTypeRefSetIds.get(0);
-		String mapTypeRefsetUri = "SNOMEDCT/" + FHIR_MAP_TYPE_REFSET_VERSION + "/" + 
-				SnomedTerminologyComponentConstants.REFSET_NUMBER + "/" + simpleMapTypeRefsetId;
+		String mapTypeRefsetUri = "SNOMEDCT/" + FHIR_MAP_TYPE_REFSET_VERSION + "/member/" + simpleMapTypeRefsetId;
```
Comment (apeteri) [inline fhir/com.b2international.snowowl.fhir.rest.tests/src/com/b2international/snowowl/fhir/rest/tests/conceptmap/SnomedConceptMapRestTest.java]: The URI should reflect that we are looking for a reference set (or a concept map if that is more FHIR-y).

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/ConceptMapCompareDsvExportTest.java [hunk 3]
```diff
@@ -153,9 +154,9 @@ private static ConceptMapCompareResultItem createItemUnspecifiedTarget(ConceptMa
 	private static ConceptMapCompareResultItem createItem(ConceptMapCompareChangeKind changeKind, String containerTerm, String sourceId, String sourceTerm, String targetId, String targetTerm) {
 		ConceptMapMapping mapping = ConceptMapMapping.builder()
 				.containerTerm(containerTerm)
-				.sourceComponentURI(ComponentURI.of(CodeSystem.uri("SNOMEDCT-CA"), (short) 100, sourceId))
+				.sourceComponentURI(ComponentURI.of(CodeSystem.uri("SNOMEDCT-CA"), SnomedConcept.TYPE, sourceId))
 				.sourceTerm(sourceTerm)
-				.targetComponentURI(ComponentURI.of(CodeSystem.uri("ICD-10-CA"), (short) 300, targetId))
+				.targetComponentURI(ComponentURI.of(CodeSystem.uri("ICD-10-CA"), SnomedConcept.TYPE, targetId))
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/ConceptMapCompareDsvExportTest.java]: This constant should refer to the ICD-10-CA-specific `concept` value, if there will be one.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedBulkRequest.java [hunk 4]
```diff
@@ -97,10 +96,10 @@ private boolean isMember(String componentId) {
...
-			case SnomedTerminologyComponentConstants.CONCEPT_NUMBER: return SnomedConceptDocument.class;
-			case SnomedTerminologyComponentConstants.DESCRIPTION_NUMBER: return SnomedDescriptionIndexEntry.class;
-			case SnomedTerminologyComponentConstants.RELATIONSHIP_NUMBER: return SnomedRelationshipIndexEntry.class;
+		switch (SnomedComponent.getType(componentId)) {
+			case SnomedConcept.TYPE: return SnomedConceptDocument.class;
+			case SnomedDescription.TYPE: return SnomedDescriptionIndexEntry.class;
+			case SnomedRelationship.TYPE: return SnomedRelationshipIndexEntry.class;
 			default: {
 				if (!isMember(componentId)) {
 					throw new UnsupportedOperationException("Cannot determine CDO class from component ID '" + componentId + "'.");
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedBulkRequest.java]: The exception message should be updated.

File: snomed/com.b2international.snowowl.snomed.fhir/src/com/b2international/snowowl/snomed/fhir/SnomedConceptMapApiProvider.java [hunk 5]
```diff
@@ -299,8 +288,8 @@ public TranslateResult translate(ComponentURI componentURI, TranslateRequest tra
 			.all()
 			.filterByActive(true)
 			.filterByTypes(CONCEPT_MAP_TYPES)
-			.filterByMapTargetComponentType(terminologyComponentIdAsInt)
-			.filterByReferencedComponentType(SnomedTerminologyComponentConstants.CONCEPT)
+			.filterByMapTargetComponentType(targetSystem)
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.fhir/src/com/b2international/snowowl/snomed/fhir/SnomedConceptMapApiProvider.java:291]: Is this going to work if `targetSystem` is not modified by the `if` statements above?

**Review by apeteri — CHANGES_REQUESTED** [review]

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/branch/compare/CompareResultsDsvExporter.java (hunk 1 above)
Comment (cmark) [reply core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/branch/compare/CompareResultsDsvExporter.java:257]: That whole class will be refactored soon, so I'll keep it as is for now.

File: snomed/com.b2international.snowowl.snomed.fhir/src/com/b2international/snowowl/snomed/fhir/SnomedConceptMapApiProvider.java (hunk 5 above)
Comment (cmark) [reply snomed/com.b2international.snowowl.snomed.fhir/src/com/b2international/snowowl/snomed/fhir/SnomedConceptMapApiProvider.java:291]: Yes, that's the plan. I've fixed the incorrect map target in this case and also the search request method infinite loop.

**Review by apeteri — APPROVED** [review]

**Pull request merged**